### DISCUSSION TITLE
DatePicker component set to null puts a random value when saved

### DISCUSSION THREAD

**Discussion by goeil** [filamentphp/filament#3534] [Help]
Hello all,

I'm experiencing a weird thing using DatePicker component.

The date field of my model is nullable.

When load this null field from DB, it displays correctly a blank field.
When I pick a date, it displays correctly and saves correctly into database.

The problem occurs when I have a non-null value, and I want to nullify it (by pressing backspace).
It saves correctly the null value in database, BUT (and here is the problem) it displays a random date value in the field after having saved.

I would like to see the date field kept blank if I just saved a null value.

            Forms\Components\DatePicker::make('birth_date')
                ->displayFormat('d mm Y')
                ->default('25 years ago')
                ->label(__('Birth date'))
                ->closeOnDateSelection(true)

I debugged in the aftertStateHydrated() function, the state is indeed null; something happens later with this field, but I cannot see what.

Do some of you experience the same issues ?

**Comment by danharrin** [comment]
Are you sure it's really random, a completely different value every time?

Please open an issue with a reproduction repo

**Reply by goeil** [reply]
No, not really random; I suppose it provides a previous entered value.
But when no value had been entered before, I got a −7 month date from today.
